src/main/java/com/google/sps/data/GetGovAdaptor.java

```diff
@@ -8,22 +8,24 @@
 import java.io.IOException;
 import java.io.OutputStream;
 import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
 
 import com.google.gson.Gson;
 import com.google.gson.reflect.TypeToken;
 
 /** File to run the GET request in java before putting it in the main adaptor function. 
  */
 public class GetGovAdaptor {
-  public static void main(final String[] args) {
-    try {
-      GetGovAdaptor.getSiteInfo();
-    } catch (final Exception e) {
-      System.out.println(e.getMessage());
-    }
+
+  public static Coordinates getCoord(HashMap<Integer, GovSiteDetails> Map, Integer siteId) {
+    return new Coordinates(Map.get(siteId).lng, Map.get(siteId).lat);
```

Comment:
Rather than having getSiteInfo() return the map, it would be better for it to store the map on a private data member (and possibly called updateSiteInfo).

This allows getCoord() to simply return the value of the stored map, and doesn't need the map to be passed back into it each time.

This would be an example of the class providing encapsulation for the site id to coord mapping. It doesn't give the raw data (ie, the map) to the "user", it handles the logic of populating and accessing it.